recipes/default.rb

```diff
@@ -15,14 +15,14 @@
   node['rackspace_users']['data_bag'],
   node['rackspace_users']['data_bag_item']).to_hash.select { |user, user_data| user != 'id' }
 
-node_tags = node['rackspace_users']['node_tags']
+node_groups = node['rackspace_users']['node_groups']
 
 groups = {}
 
 users.each do |username, user_data|
-  ## The action (if defined in the data bag) should be overwritten to 'remove' if the user's node tags and node's tags don't intersect.
-  ## If the user doesn't declare any tags then it is assumed that is present on all nodes.
-  user_data['action'] = 'remove' if user_data['node_tags'] && (user_data['node_tags'] & node_tags).empty?
+  ## The action (if defined in the data bag) should be overwritten to 'remove' if not all the user's node_groups are in node's node_groups.
+  ## If the user doesn't declare node_groups then it is assumed that is present on all nodes.
+  user_data['action'] = 'remove' if user_data['node_groups'] && !(user_data['node_groups'] - node_groups).empty?
```

Comment:
I think there is value in having both and I don't think it will be overly complicated to have an attribute that flips this behaviour from 'any' to 'all' with 'any' being the default if you prefer. I can see use cases for both, it's just that with 'any' you cannot do more strict memberships. I will leave this for another day in case someone wants to comment and I can implement this logic along with examples in the README.